app/controllers/variable_drafts_controller.rb

```diff
@@ -13,6 +14,45 @@ def new
     set_current_form
   end
 
+  def show
+    super
+
+    validate_metadata
+  end
+
+  def publish
```

Comment:
In an effort to use rails conventions as often as possible, it seems as though we could use `create` instead of `publish` which would allow us to use the standard 7 actions within rails.